modules/sdk-coin-avaxc/src/avaxc.ts

```diff
@@ -1180,4 +1191,51 @@ export class AvaxC extends BaseCoin {
   getAvaxP(): string {
     return this.getChain().toString() === 'avaxc' ? 'avaxp' : 'tavaxp';
   }
+
+  /**
+   * Fetch the gas price from the explorer
+   */
+  async getGasPriceFromExternalAPI(): Promise<BN> {
+    try {
+      const res = await this.recoveryBlockchainExplorerQuery({
+        jsonrpc: '2.0',
+        method: 'eth_gasPrice',
+        id: 1,
+      });
+      const gasPrice = new BN(res.result.slice(2), 16);
+      console.log(` Got gas price: ${gasPrice}`);
+      return gasPrice;
+    } catch (e) {
+      throw new Error('Failed to get gas price');
+    }
+  }
+
+  /**
+   * Fetch the gas limit from the explorer
+   * @param from
+   * @param to
+   * @param data
+   */
+  async getGasLimitFromExternalAPI(from: string, to: string, data: string): Promise<BN> {
+    try {
+      const res = await this.recoveryBlockchainExplorerQuery({
+        jsonrpc: '2.0',
+        method: 'eth_estimateGas',
+        params: [
+          {
+            from,
+            to,
+            data,
+          },
+          'latest',
+        ],
+        id: 1,
+      });
+      const gasLimit = new BN(res.result.slice(2), 16);
+      console.log(`Got gas limit: ${gasLimit}`);
+      return gasLimit;
+    } catch (e) {
+      throw new Error('Failed to get gas limit: ');
```

Comment:
We already have these functions in the abstract class. Since the way we fetch the gas values from explorer for AVAXC is different, we have overriden these functions in AVAXC class.